common/rpc/interceptor/namespace_validator.go

```diff
@@ -237,10 +243,41 @@ func (ni *NamespaceValidatorInterceptor) StateValidationIntercept(
 // 4. Namespace from request match namespace from task token, if check is enabled with dynamic config.
 // 5. Namespace is in correct state.
 func (ni *NamespaceValidatorInterceptor) ValidateState(namespaceEntry *namespace.Namespace, fullMethod string) error {
-	if err := ni.checkNamespaceState(namespaceEntry, fullMethod); err != nil {
+	return ni.checkNamespaceState(namespaceEntry, fullMethod)
+}
+
+func (ni *NamespaceValidatorInterceptor) handleRequestWithHandoverRetry(
+	ctx context.Context,
+	req interface{},
+	info *grpc.UnaryServerInfo,
+	handler grpc.UnaryHandler,
+	nsEntry *namespace.Namespace,
+) (interface{}, error) {
+
+	var response interface{}
+	var retryCount int
+	retryLatency := time.Duration(0)
+	op := func(ctx context.Context) error {
+		err := ni.checkReplicationState(nsEntry, info.FullMethod)
+		if err != nil {
+			return err
+		}
+
+		response, err = handler(ctx, req)
+		if retryCount > 1 {
+			retryLatency = retryLatency + handoverRetryPolicy.ComputeNextDelay(0, retryCount, nil)
+		}
+		retryCount++
```

Comment:
it's a bit hard to follow as it's calculating the retryLatency before the current attempt, but the code lives after the handler call...